smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -65,6 +70,131 @@ void writeMembers(TypeScriptWriter writer, Shape shape) {
+                () -> {
+                    MemberShape mapMember = ((MapShape) memberShape).getValue();
+                    writeFilterSensitiveLogForMap(writer, mapMember);
+                }
+            );
+        } else {
+            // This path will never reach because of recursive isIterationRequired
+            // adding it to not break the code, if it does reach in future
+            writer.write("item => item");
+        }
+    }
+
+    /**
+     * Recursively writes filterSensitiveLog for MapShape
+     */
+    void writeFilterSensitiveLogForMap(TypeScriptWriter writer, MemberShape mapMember) {
+        // Reducer is common to all shapes
+        writer.openBlock("(acc: any, [key, value]: [string, ${T}]) => {", "}, {}",
+            symbolProvider.toSymbol(mapMember),
+            () -> {
+                Shape memberShape = model.expectShape(mapMember.getTarget());
+                if (memberShape instanceof StructureShape) {
+                    // Call filterSensitiveLog on Structure
+                    writer.write("acc[key] = ${T}.filterSensitiveLog(value);",
+                        symbolProvider.toSymbol(mapMember));
+                } else if (memberShape instanceof CollectionShape) {
+                    writer.openBlock("acc[key] = value.map(", ")",
+                        () -> {
+                            MemberShape collectionMember = ((CollectionShape) memberShape).getMember();
+                            writeFilterSensitiveLogForCollection(writer, collectionMember);
+                        }
+                    );
+                } else if (memberShape instanceof MapShape) {
+                    writer.openBlock("acc[key] = Object.entries(value).reduce(", ")",
+                        () -> {
+                            MemberShape nestedMapMember = ((MapShape) memberShape).getValue();
+                            writeFilterSensitiveLogForMap(writer, nestedMapMember);
+                        }
+                    );
+                } else {
+                    // This path will never reach because of recursive isIterationRequired
+                    // adding it to not break the code, if it does reach in future
+                    writer.write("acc[key] = value;");
+                }
+                writer.write("return acc;");
+            }
+        );
+    }
+
+    void writeFilterSensitiveLog(TypeScriptWriter writer, Shape shape) {
+        writer.write("...obj,");
+        for (MemberShape member : members) {
+            Shape memberShape = model.expectShape(member.getTarget());
+            String memberName = TypeScriptUtils.sanitizePropertyName(symbolProvider.toMemberName(member));
+            if (member.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
+                // member is Sensitive, hide the value
+                writer.write("...(obj.${L} && { ${L}: SENSITIVE_STRING }),", memberName, memberName);
```

Comment:
There are a chunk of places where you can make use of [positional parameters](https://github.com/awslabs/smithy/blob/master/smithy-utils/src/main/java/software/amazon/smithy/utils/CodeWriter.java#L113:L127) in your `.write()` calls.

```java
writer.write("...(obj.$1L && { $1L: SENSITIVE_STRING }),", memberName);
```

Reply:
Updated in https://github.com/awslabs/smithy-typescript/pull/170/commits/f84179a8a70016ae31834c4a51bcb7b2e5367ffb